editoast/src/views/work_schedules.rs

```diff
@@ -275,6 +294,226 @@ async fn project_path(
+        .await?
+        .into_iter()
+        .map(|work_schedule| work_schedule.id);
+
+    // Delete them
+    use crate::models::DeleteBatch;
+    let conn = &mut db_pool.get().await?;
+    WorkSchedule::delete_batch(conn, work_schedule_ids).await?;
+
+    // Delete the group itself
+    WorkScheduleGroup::delete(&work_schedule_group, conn).await?;
+
+    Ok(axum::http::StatusCode::NO_CONTENT)
+}
+
+#[utoipa::path(
+    get, path = "",
+    tag = "work_schedules",
+    responses(
+        (status = 201, body = Vec<i64>, description = "The existing work schedule group ids"),
+    )
+)]
+async fn list_groups(
+    State(app_state): State<AppState>,
+    Extension(auth): AuthenticationExt,
+) -> Result<Json<Vec<i64>>> {
+    let authorized = auth
+        .check_roles([BuiltinRole::WorkScheduleRead].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Unauthorized.into());
+    }
+
+    let db_pool = app_state.db_pool_v2.clone();
+    let conn = &mut db_pool.get().await?;
+
+    let selection_setting = SelectionSettings::new();
+    let work_schedule_group_ids = WorkScheduleGroup::list(conn, selection_setting)
+        .await?
+        .iter()
+        .map(|group| group.id)
+        .collect::<Vec<i64>>();
+
+    Ok(Json(work_schedule_group_ids))
+}
+
+#[derive(Serialize, Deserialize, ToSchema)]
+struct WorkScheduleInGroupCreateForm {
+    work_schedules: Vec<WorkScheduleItemForm>,
+}
+
+#[utoipa::path(
+    put, path = "",
+    tag = "work_schedules",
+    request_body = Vec<WorkScheduleItemForm>,
+    params(WorkScheduleGroupIdParam),
+    responses(
+        (status = 200, description = "The work schedules have been created", body = Vec<WorkSchedule>),
+        (status = 404, description = "Work schedule group not found"),
+    )
+)]
+async fn put_in_group(
+    State(app_state): State<AppState>,
+    Extension(auth): AuthenticationExt,
+    raw_group_id: Path<WorkScheduleGroupIdParam>,
+    Json(work_schedules): Json<Vec<WorkScheduleItemForm>>,
+) -> Result<Json<Vec<WorkSchedule>>> {
+    let authorized = auth
+        .check_roles([BuiltinRole::WorkScheduleWrite].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Unauthorized.into());
+    }
+    let group_id = raw_group_id.id;
+
+    let db_pool = app_state.db_pool_v2.clone();
+    let conn = &mut db_pool.get().await?;
+
+    // Check that the group exists
+    WorkScheduleGroup::retrieve_or_fail(conn, group_id, || {
+        WorkScheduleError::WorkScheduleGroupNotFound { id: group_id }
+    })
+    .await?;
+
+    // Create work schedules
+    let work_schedules_changesets = work_schedules
+        .into_iter()
+        .map(|work_schedule| work_schedule.into_work_schedule_changeset(group_id))
+        .collect::<Vec<_>>();
+    let work_schedules = WorkSchedule::create_batch(conn, work_schedules_changesets).await?;
```

Comment:
Same here, maybe we need a transaction, because I'm not sure what happens if the work schedule group is deleted in between?

Reply:
Fixed in f937a335fad70ebc83e5bc3a0992a73e553d2b67

It would have created "orphaned" work schedules that couldn't be used, queried, or deleted